react.md

```diff
@@ -0,0 +1,242 @@
+  <ComponentName x={@state.x}/> # Bad
+  ```
+
+4. Wrap all inline object literals in curly braces
+
+  ```CoffeeScript
+  <ComponentName x={y: 5} /> # Bad
+  <ComponentName x={{y: 5}} /> # Good
+  ```
+
+5. If you are implicitly passing down `props` or `state`, use the Spread operator:
+
+  ```CoffeeScript
+  # @state = {x: 5, y: 6, z: 7}
+  <ComponentName {...@state} /> # Then you can call `@prop.x` inside of `ComponentName`
+  ```
+
+6. Use multi-line if-statements wherever possible
+
+  ```CoffeeScript
+  render: ->
+    text =
+      if @state.isVisible
+        <span>Or this is alright, too</span>
+    <span>
+      {text} 
+      {if @state.isVisible then <h1>Show this</h1>} # Good, since it's short
+      {if @state.isVisible then <p><strong>Bold first word</strong> and regular 
+       elsewhere</p>} # Bad, because it's not short
+      {if @state.isVisible then {@state.text}} # Good, since it's short (and a variable)
+      {if @state.isVisible
+        <p><strong>Bold first word</strong> and regular stuff elsewhere</p>
+      } # Good, since it's long
+    </span>
+  ```
+
+7. If you have multiple mixins, define each one on a new line:
+
+  ```CoffeeScript
+  SomeComponent = React.createClass
+    mixins: [
+      MixinOne
+      MixinTwo
+    ]
+  ```
+
+8. Remember to `key` like elements that appear in arrays
+  1. Any adjacent React Components are represented by arrays of elements.
+  2. This is to provide React with the information to hueristically optimize UI reconciliation (figuring out what to re-render and what not to)
+
+  ```CoffeeScript
+  render: ->
+    <div>
+      <p key={1}>Some text</p>
+      <p key={2}>Some other text</p>
+      <p key={3}>Some more text</p>
+    </div>
+
+  makeItem: (item, index) ->
+    <p key={index}>{item}</p> # `map`-ing over the elements
+                              # lets you use the index of the item
+                              # in the array as a key.
+                              # 
+                              # `key` just has to be unique in the array,
+                              # it doesn't have to be unique across the page.
+                              #
+                              # If you are constantly re-sorting items in the
+                              # array, try to use the resource's ID. If not,
+                              # using the index of the element in the source array works
+  render: ->
+    <div>
+      {@state.items.map(@makeItem)}
+    </div>
+  ```
+
+9. Use expressions to iterate over datasets & render components
+  1. `map`, `filter`, `concat`, etc.
+  2. Lodash/underscore methods work as well (`where`, `reject`, `pluck`, etc.)
+
+  ```Coffeescript
+  ###
+  @props.items = [1,2,3,4,5]
+  ###
+
+  # Good example
+  showEvenItems: (item, index) ->
+    item % 2 == 0
+  makeItem: (item, index) ->
+    <li key={item.database_id}>{item}</li>
+  render: ->
+    <ul>
+      {@props.items.filter(@showEvenItems).map(@makeItem)}
```

Comment:
Current consensus: Use `lodash` methods for now to ensure browser compatibility.